### PR TITLE
feat(datepicker): add monthColumns to DatepickerConfig

### PR DESCRIPTION
**Pull request by Mrman**
#5515

### CODE REVIEW

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -458,11 +461,13 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
 }])
 
 .controller('UibMonthpickerController', ['$scope', '$element', 'dateFilter', function(scope, $element, dateFilter) {
+  var columns;
   this.step = { years: 1 };
   this.element = $element;
 
   this.init = function(ctrl) {
     angular.extend(ctrl, this);
+    columns = ctrl.monthColumns;
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Why does this need to be stored - couldn't one just use `ctrl.monthColumns` when needed?

File: src/datepicker/datepicker.js [hunk 2]
```diff
@@ -55,6 +56,7 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
     'showWeeks',
     'shortcutPropagation',
     'startingDay',
+    'monthColumns',
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Move this up to preserve alphabetical order

File: src/datepicker/datepicker.js [hunk 3]
```diff
@@ -79,6 +81,7 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
         break;
       case 'showWeeks':
       case 'shortcutPropagation':
+      case 'monthColumns':
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Ditto on this for alphabetical ordering

File: src/datepicker/docs/readme.md [hunk 4]
```diff
@@ -123,6 +123,11 @@ Apart from the previous settings, to configure the uib-datepicker you need to cr
     <small class="badge">C</small>
     *(Default: `$locale.DATETIME_FORMATS.FIRSTDAYOFWEEK`)* -
     Starting day of the week from 0-6 (0=Sunday, ..., 6=Saturday).
+    
+  * `monthColumns`
```
Comment (wesleycho) [inline src/datepicker/docs/readme.md]: Move section up to preserve alphabetical ordering

File: src/datepicker/test/datepicker.spec.js [hunk 5]
```diff
@@ -1346,13 +1351,17 @@ describe('datepicker', function() {
...
-          ['Apr', 'May', 'Jun'],
-          ['Jul', 'Aug', 'Sep'],
-          ['Oct', 'Nov', 'Dec']
+          ['Jan', 'Feb', 'Mar', 'Apr'],
+          ['May', 'Jun', 'Jul', 'Aug'],
+          ['Sep', 'Oct', 'Nov', 'Dec']
         ]);
       });
 
+      it('changes colspan of yearbutton in month selector to have colspan', function() {
```
Comment (wesleycho) [inline src/datepicker/test/datepicker.spec.js]: Test name should be descriptive on what is happening at a high level, not specifics about implementation

**Comment by Mrman** [comment]
Thanks for review comments. Have updated the PR with the changes.

**Comment by icfantv** [comment]
@wesleycho, what needed work on this?  is it good to merge or do we need more info?

**Comment by Mrman** [comment]
Would love to get this into the next release. Do let me know if anything is outstanding. Addressed all the comments raised in the initial review.

**Comment by wesleycho** [comment]
Sorry for the long wait - this LGTM

**Pull request closed without merging**